webapp/src/Entity/Problem.php

```diff
@@ -287,26 +283,82 @@ public function getSpecialCompareArgs(): ?string
         return $this->special_compare_args;
     }
 
-    public function setCombinedRunCompare(bool $combinedRunCompare): Problem
+    public function setTypes(array $types): Problem
     {
-        $this->combined_run_compare = $combinedRunCompare;
+        $types = array_unique($types);
+
+        $this->types = 0;
+        foreach ($types as $type) {
+            if ($type === 'pass-fail') {
+                $this->types |= self::TYPE_PASS_FAIL;
+            } elseif ($type === 'scoring') {
+                $this->types |= self::TYPE_SCORING;
+            } elseif ($type === 'multi-pass') {
+                $this->types |= self::TYPE_MULTI_PASS;
+            } elseif ($type === 'interactive') {
+                $this->types |= self::TYPE_INTERACTIVE;
+            }
```

Comment:
I think this should raise an exception on a fall through case.

Also, why not use `switch ... case` here?

Reply:
reworked the code according to your second suggestion below, also added exceptions